implementation/src/test/java/io/smallrye/health/SmallRyeHealthReporterTest.java

```diff
@@ -173,4 +177,14 @@ public void testGetHealthWithMixedChecksAndStyleStackTrace() {
         assertThat(health.getPayload().getJsonArray("checks").getJsonObject(2).getString("name"), is("down"));
         assertThat(health.getPayload().getJsonArray("checks").getJsonObject(2).getString("status"), is("DOWN"));
     }
+
+    @Test
+    public void testReportWhenDown() {
+        reporter.addHealthCheck(new DownHealthCheck());
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+
+        reporter.reportHealth(bos, reporter.getHealth());
+
+        assertThat(bos.toString(), is(not(emptyString())));
```

Comment:
This will return the payload in `bos` so you are not checking for the log message. The content of `bos` will be a JSON with the failed `DownHealthCheck`.

Reply:
I initially wrote the test to run in IDE and manually checked the console.
But I've found a way to assert logs via an extra handler on the JDK logger used behind jboss-logging.